tla-types/src/main/scala/at/forsyte/apalache/tla/typecomp/signatures/FiniteSetOperSignatures.scala

```diff
@@ -0,0 +1,31 @@
+package at.forsyte.apalache.tla.typecomp.signatures
+
+import at.forsyte.apalache.tla.lir.oper.TlaFiniteSetOper
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.typecomp.{BuilderUtil, SignatureMap}
+
+/**
+ * Produces a SignatureMap for all set operators
+ *
+ * @author
+ *   Jure Kukovec
+ */
+object FiniteSetOperSignatures {
+  import BuilderUtil._
+  import TlaFiniteSetOper._
+
+  /**
+   * Returns a map that assigns a signature generator to each TlaSetOper. Because most operators are polymorphic, their
+   * signatures will contain type variables produced on-demand by varPool.
+   */
+  def getMap: SignatureMap = {
+
+    // (Set(t)) => Bool
+    val isFSSig = signatureMapEntry(isFiniteSet, { case Seq(SetT1(_)) => BoolT1 })
+
+    // (Set(t)) => Int
+    val cardSig = signatureMapEntry(cardinality, { case Seq(SetT1(_)) => IntT1 })
```

Comment:
```suggestion
    val cardinalitySig = signatureMapEntry(cardinality, { case Seq(SetT1(_)) => IntT1 })
```

because the operator is called `Cardinality`